test/greeter/Greeter.ts

```diff
@@ -1,24 +1,38 @@
+import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
 import type { SignerWithAddress } from "@nomiclabs/hardhat-ethers/dist/src/signer-with-address";
-import { artifacts, ethers, waffle } from "hardhat";
-import type { Artifact } from "hardhat/types";
+import { ethers } from "hardhat";
 
+import { Greeter__factory } from "../../src/types";
```

Comment:
This should use "import type" and be imported from "../../src/types/factories/Greeter__factory".

Reply:
done